streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/reprojection/ProjTransformation.java

```diff
@@ -0,0 +1,138 @@
+                //TODO implement fallback option with proj4j
+                throw new SpRuntimeException("Database not set and ready for fallback ");
+            }
+        } catch (FactoryException e) {
+            throw new SpRuntimeException("Something unexpected happened " + e);
+        }
+
+        // check if SIS DB has the supported 9.9.1 Version.
+        try {
+            if (!SpReprojectionBuilder.isSisDbCorrectVersion()) {
+                logger.warn("Not supported EPSG DB used.");
+                throw new SpRuntimeException("Your current EPSG DB version " + SpReprojectionBuilder.getSisDbVersion()
+                        + " is not the supported 9.9.1 version. ");
+            }
+        } catch (FactoryException e) {
+            throw new SpRuntimeException("Something unexpected happened " + e);
+        }
+
+        // checks if Input EPSG in valid and exists in EPSG DB
+        if (!SpReprojectionBuilder.isSisEpsgValid(targetEPSG)) {
+            throw new SpRuntimeException("Your chosen EPSG Code " + targetEPSG + " is not valid or supported. "
+                    + "Check EPSG on https://spatialreference.org");
+        }
+    }
+
+    @Override
+    public void onEvent(Event in, SpOutputCollector out) {
+
+        String wkt = in.getFieldBySelector(params.getWktString()).getAsPrimitive().getAsString();
+        Integer epsgCode = in.getFieldBySelector(params.getEpsgCode()).getAsPrimitive().getAsInt();
+        Geometry geometry = SpGeometryBuilder.createSPGeom(wkt, epsgCode);
+
+        Geometry transformed = null;
+        try {
+            transformed = SpReprojectionBuilder.reprojectSpGeometry(geometry, targetEPSG);
+        } catch (SpNotSupportedGeometryException e) {
+            transformed = SpGeometryBuilder.createEmptyGeometry(geometry);
+        }
+
+        if (!transformed.isEmpty()) {
+            in.updateFieldBySelector("s0::" + ProjTransformationController.EPSG_RUNTIME, params.getTargetEpsg());
+            in.updateFieldBySelector("s0::" + ProjTransformationController.WKT_RUNTIME, transformed.toText());
+
+            out.collect(in);
+        } else {
+            logger.warn("An empty point geometry is created in " + ProjTransformationController.EPA_NAME + " " +
+                    "due invalid input values. Check used epsg Code:" + epsgCode);
+        }
+    }
+
+    @Override
+    public void onDetach() {
+    }
+
+    // https://sis.apache.org/apidocs/org/apache/sis/setup/Configuration.html#setDatabase(java.util.function.Supplier)
+    // TODO: Best would be ConfigKeys VARIABLE. ATM hardcoded and adjustments for development required like IP of client
+    // TODO: has to move together with the Configuration.current() method
+    protected static DataSource createDataSource() {
+        PGSimpleDataSource ds = new PGSimpleDataSource();
+        // HAS TO BE ADJUSTED OR INCLUDED IN THE AUTO_DISCOVERY
+        String[] serverAddresses = {"192.168.178.100"};
```

Comment:
Can we replace this internal IP with an environment variable?

Reply:
Can you show me an example with the consul variables please?
BTW is it save to put this and the cretaConenction in the GeoJvmInit ?

This  method  just has to be called  ones during "start"
So it can be used by all processors.